### ISSUE TITLE
ACF Image Field Rendering

### ISSUE THREAD

**Issue by stevebab** [humanmade/network-media-library#34]
Incredibly helpful plugin. Thank you for your exceptional work and foresight. 

It seems that ACF image fields are using the id of the current blog rather than the network media source, and therefore return an empty array to the theme layer. 

I noticed that in version 1.2.0 that there is code about line 421 that forces galleries to use the network media source id. Would this be adaptable to also work with ACF?

**Comment by johnbillion** [comment]
ACF support is one of the items on my todo list. I'm hoping that ACF has some filters in it that facilitate serving its media files from the central media site. It should be possible.

See #21.

**Comment by bedrijfsportaal** [comment]
Hello,

Thank you for the incredibly helpful plugin, we still have issues regarding the ACF field rendering in combination with Gutenberg editor. 

When debugging the information below is generated with a HTTP 500 Error;
`post.php?post=5&action=edit&meta-box-loader=1&_wpnonce=82cecdefef&_locale=user | 500 `

When deactivating everything is working as expected.

**Comment by johnbillion** [comment]
@bedrijfsportaal Are you able to check your PHP error logs to find out what's causing the 500? It might be a fatal error.

**Comment by fine-G** [comment]
in the image crop and file field it is possible to add switch_blog() to bring them to work!

**Comment by raelshark** [comment]
Came to say that I successfully used `switch_to_blog()` as mentioned above to get an ACF image field's content from my main site.

Here's a full example from inside a repeater field:

```
switch_to_blog(1);
$image = get_sub_field('photo');
restore_current_blog();
```

However, I'm currently stuck trying to get the same to work with an ACF options page image field. Using `switch_to_blog()` in that context gets back the corresponding option field for the switched blog - which is more of an ACF issue, but means the shared image can't be retrieved from the original source this way.

**Comment by raelshark** [comment]
Okay, I figured out how to grab ACF images that are stored as options. The problem is in the "format" parameter on `get_field()` - it defaults to "true", which causes ACF to call `get_post()` on the image's ID, which we don't want because it looks for the image in the wrong blog.

So I was able to get it working by passing "false" for the "format" argument to `get_field()`, which gets back the image's ID, which we can then use to call `acf_get_attachment()` in the proper blog;

```php
$image_id = get_field('photo', 'option', false);
switch_to_blog( $main_blog );
$image = acf_get_attachment( $image_id, 'full' );
restore_current_blog();
```
Not sure if that's relevant to getting this working in NML, but hopefully it's helpful to someone.

**Comment by stevebab** [comment]
Thanks for the post @raelshark

**Comment by johnbillion** [comment]
Fixed in 1.5.0.

**Comment by johnbillion** [comment]
Thanks for those code pointers @raelshark, it helped a lot. Cheers!
